### DISCUSSION TITLE
Need Help In Fixing Prime React Datatable (Search Input Lag)

### DISCUSSION THREAD

**Discussion by abdulrehmanwaseem** [community/community#140311] [Programming Help] [answered]
### Body

Hi all, 👋

I’m currently facing an issue with the search functionality in Prime React DataTable where there's a noticeable lag when typing in the search input field. The table contains a considerable amount of data, and every keystroke results in a delayed response, making the search experience quite slow.

What I’ve Tried So Far:
Debouncing the search input: I’ve implemented debouncing using lodash.debounce to minimize the frequency of state updates. This improved the performance slightly, but there’s still a lag.

Pagination and lazy loading: I enabled pagination and lazy loading for the data, but the search is still sluggish even with smaller data sets.

React.memo and useMemo: I tried wrapping components with React.memo and using useMemo for memoizing expensive calculations, but the improvement was minimal.

Environment:
Prime React version: 10.6.6
React version: 18.3.1
Using functional components with hooks
My Question:
Does anyone have experience optimizing the search functionality in Prime React’s DataTable? Are there any additional techniques or best practices for improving the performance of the search input, especially with large datasets?

Any help or suggestions would be greatly appreciated! 🙏

Thanks in advance!
Abdul Rehman

### Guidelines

- [x] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by mohdsaadshaikh** [comment]
Hey Abdul,

It sounds like you’ve already tried some good optimization techniques! I’ve run into similar issues with third-party libraries in the past, and sometimes the best course of action is to reach out directly to PrimeReact’s customer support or submit a bug report on their official GitHub repository.

They might already be aware of the issue and could have a patch or workaround for it. You could also check their forums or documentation for any specific guidance on optimizing performance with large datasets.

Hope this helps! Let us know if you find a solution.

Best of luck!

**Comment by abdulrehmanwaseem** [comment]
Got The Answer By mohdsaadshaikh

**Comment by ebndev** [comment, marked as answer]
Please see [this comment](https://github.com/orgs/community/discussions/140314#discussioncomment-10824381) for reference to why this discussion was closed and locked. Thanks!
